open_spiel/python/mfg/examples/mfg_Proximal_policy_optimization_pytorch.py

```diff
@@ -0,0 +1,448 @@
+    parser.add_argument("--num-minibatches", type=int, default=5,  help="the number of mini-batches")
+    parser.add_argument("--update-epochs", type=int, default=5, help="the K epochs to update the policy")
+    parser.add_argument("--clip-coef", type=float, default=0.2, help="the surrogate clipping coefficient")
+    parser.add_argument("--ent-coef", type=float, default=0.01, help="coefficient of the entropy")
+    parser.add_argument("--max-grad-norm", type=float, default=5, help="the maximum norm for the gradient clipping")
+
+
+    # MFPPO parameters
+    parser.add_argument('--alpha', type= int, default=0.5, help='Set alpha to controll the iteration and epsiode policy updates')
+    parser.add_argument('--eps-eps', type= int, default=0.2, help='eps to update the episode learned policy')
+    parser.add_argument('--itr-eps', type= int, default=0.05, help='eps to update the episode learned policy')
+
+    args = parser.parse_args()
+
+    return args
+
+
+class NashC(NashConv):
+    # Mainly used to calculate the exploitability 
+    def __init__(self, game,distrib,pi_value, root_state=None):
+        self._game = game
+        if root_state is None:
+            self._root_states = game.new_initial_states()
+        else:
+            self._root_states = [root_state]
+            
+        self._distrib = distrib
+
+        self._pi_value = pi_value
+
+        self._br_value = best_response_value.BestResponse(
+            self._game,
+            self._distrib,
+            value.TabularValueFunction(self._game),
+            root_state=root_state)
+
+
+class Agent(nn.Module):
+    def __init__(self, info_state_size, num_actions):
+        super(Agent, self).__init__()
+        self.num_actions = num_actions
+        self.info_state_size = info_state_size
+        self.critic = nn.Sequential(
+            layer_init(nn.Linear(info_state_size, 128)), 
+            nn.Tanh(),
+            layer_init(nn.Linear(128,128)),
+            nn.Tanh(),
+            layer_init(nn.Linear(128,1))
+        )
+        self.actor = nn.Sequential(
+            layer_init(nn.Linear(info_state_size, 128)),
+            nn.Tanh(),
+            layer_init(nn.Linear(128,128)),
+            nn.Tanh(),
+            layer_init(nn.Linear(128, num_actions))
+        )
+        
+
+    def get_value(self, x):
+        return self.critic(x)
+
+    def get_action_and_value(self, x, action=None):
+        logits = self.actor(x)
+        probs = Categorical(logits=logits)
+        if action is None:
+            action = probs.sample()
+        return action, probs.log_prob(action), probs.entropy(), self.critic(x)
+
+
+def layer_init(layer, bias_const=0.0):
+    # used to initalize layers
+    nn.init.xavier_normal_(layer.weight)
+    nn.init.constant_(layer.bias, bias_const)
+    return layer
+
+
+class PPOpolicy(policy_std.Policy):
+    # required obeject to work with openspiel 
```

Comment:
Use the Google3 Python style doc strings.

See e.g.: https://github.com/google-deepmind/open_spiel/blob/master/open_spiel/python/mfg/algorithms/average_network_fictitious_play.py

Reply:
Dockstrings were added into the code